target/linux/ath79/dts/qca9558_engenius_emr3000-v2.dts

```diff
@@ -0,0 +1,155 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+
+#include "qca955x.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	model = "EnGenius EMR3000 V2";
+	compatible = "engenius,emr3000-v2", "qca,qca9558";
+
+	aliases {
+		led-boot = &led_status_white;
+		led-failsafe = &led_status_white;
+		led-running = &led_status_white;
+		led-upgrade = &led_status_white;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_status_white: status_white {
+			label = "white:status";
+			gpios = <&gpio 18 GPIO_ACTIVE_HIGH>;
+			default-state = "on";
+		};
+
+		status_red {
+			label = "red:status";
+			gpios = <&gpio 11 GPIO_ACTIVE_HIGH>;
+		};
+
+		status_amber {
+			label = "amber:status";
+			gpios = <&gpio 16 GPIO_ACTIVE_HIGH>;
+		};
+
+		bluetooth {
+			label = "blue:bluetooth";
+			gpios = <&gpio 19 GPIO_ACTIVE_HIGH>;
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "reset";
+			linux,code = <KEY_RESTART>;
+			gpios = <&gpio 17 GPIO_ACTIVE_LOW>;
+		};
+	};
+};
+
+&mdio0 {
+	status = "okay";
+
+	phy0: ethernet-phy@0 {
+		reg = <1>;
+		phy-mode = "rgmii";
+	};
+
+	phy1: ethernet-phy@1 {
+		reg = <2>;
+		phy-mode = "sgmii";
+		at803x-override-sgmii-link-check;
+	};
+};
+
+&eth0 {
+	status = "okay";
+
+	phy-handle = <&phy0>;
+	pll-data = <0x9a000000 0x80000101 0x80001313>;
+
+	gmac-config {
+		device = <&gmac>;
+		rgmii-enabled = <1>;
+	};
+};
+
+&eth1 {
+	status = "okay";
+
+	phy-handle = <&phy1>;
+	pll-data = <0x9b000000 0x80000101 0x80001313>;
+	qca955x-sgmii-fixup;
+};
+
+&pcie0 {
+	status = "okay";
+};
+
+&spi {
+	status = "okay";
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <25000000>;
```

Comment:
It is very likely that frequency can be bumped to at least 40 MHz
`		spi-max-frequency = <40000000>;`

you can verify by reading datasheet of the flash chip
also the flash model should be in commit message